plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -364,6 +360,60 @@ fu_plugin_thunderbolt_find_nvmem (GUdevDevice  *udevice,
 }
 
 static gboolean
+fu_plugin_thunderbolt_validate_firmware (GUdevDevice  *udevice,
+					 GBytes       *blob_fw,
+					 GError      **error)
+{
+	g_autoptr(GFile) nvmem = NULL;
+	g_autoptr(GInputStream) is = NULL;
+	g_autoptr(GByteArray) controller_fw = g_byte_array_new();
+
+	nvmem = fu_plugin_thunderbolt_find_nvmem (udevice, TRUE, error);
+	if (nvmem == NULL)
+		return FALSE;
+
+	is = (GInputStream *) g_file_read (nvmem, NULL, error);
```

Comment:
If you want to read the whole file, you could just use `g_file_load_contents`.

Reply:
Oh, nice! Missed it.
(It's my first time to use GTK libs, as much as I remember. Any feedback on the correct way to use it would be very welcome!)